webhook/icinga.go

```diff
@@ -185,22 +186,34 @@ func updateOrCreateService(icinga icinga2.Client,
 
 	status := severityToExitStatus(alert.Status, alert.Labels["severity"], c.GetConfig().MergedSeverityLevels)
 
-	serviceData := createServiceData(hostname, serviceName, displayName, alert, status, heartbeatInterval, c)
+	serviceData, serviceVars := createServiceData(hostname, serviceName, displayName, alert, status, heartbeatInterval, c)
 
 	icingaSvc, err := icinga.GetService(serviceData.FullName())
 	// update or create service, depending on whether object exists
 	if err == nil {
 		l.Infof("updating service: %+v\n", icingaSvc.Name)
+
+		// Templates needs to be ignored if the service is already created due to the Error:
+		// Attribute 'templates' could not be set: Error: Attribute cannot be modified.
+		serviceData.Templates = nil
+
 		err := icinga.UpdateService(serviceData)
 		if err != nil {
 			return serviceData, err
 		}
```

Comment:
I haven't tested this yet, but I assume we also need to call `icinga.UpdateCustomVars(serviceData.FullName(), serviceVars)` when updating the service.